test/any/FetchWebIDLFile-test.js

```diff
@@ -0,0 +1,113 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('FetchWebIDLFile', function() {
+  var IDLFile;
+  var GithubIDLFile;
+  var GitilesIDLFile;
+  var payloadContent = 'Some sort of file content';
+
+  beforeAll(function() {
+    IDLFile = foam.lookup('org.chromium.webidl.IDLFile');
+    GithubIDLFile = foam.lookup('org.chromium.webidl.GithubIDLFile');
+    GitilesIDLFile = foam.lookup('org.chromium.webidl.GitilesIDLFile');
+  });
+
+  beforeEach(function() {
+    foam.CLASS({
+      package: 'org.chromium.webidl.test',
+      name: 'MockHTTPRequest',
+      extends: 'foam.net.HTTPRequest',
+
+      methods: [
+        function send() {
+          if (this.url == 'http://test.url/someFile.idl') {
```

Comment:
Let's make this data. E.g., class has `constants: { ...}` such as `simpleTestURL`, `chromiumTestURL`, etc., and then also has `urlMap` property with `factory` that initializes with:

```js
  var map = {};
map[this.SIMPLE_TEST_URL] = this.HTTPResponse.create({status: ..., payload: ...});
/*...*/
return map;
```

Then `send()` simply delivers pre-selected responses, or `Promise.reject(...)`s when `this.urlMap[this.url]` is `undefined`.